scipy/stats/_odds_ratio.py

```diff
@@ -0,0 +1,479 @@
+    "unexposed" to some factor or treatment. If the occurrence of some
+    illness is under study, those who have the illness are often
+    classifed as "cases", and those without it are "noncases".  The
+    counts of the occurrences of these classes gives a contingency
+    table::
+
+                    exposed    unexposed
+        cases          a           b
+        noncases       c           d
+
+    The sample odds ratio may be written ``(a/c) / (b/d)``.  ``a/c`` can
+    be interpreted as the odds of a case occurring in the exposed group,
+    and ``b/d`` as the odds of a case occurring in the unexposed group.
+    The sample odds ratio is the ratio of these odds.  If the odds ratio
+    is greater than 1, it suggests that there is a positive association
+    between being exposed and being a case.
+
+    Interchanging the rows or columns of the contingency table inverts
+    the odds ratio, so it is import to understand the meaning of labels
+    given to the rows and columns of the table when interpreting the
+    odds ratio.  This is in contrast to the p-value, which is invariant
+    under interchanging the rows or columns.
+
+    Consider a hypothetical example where it is hypothesized that
+    exposure to a certain chemical is assocated with increased occurrence
+    of a certain disease.  Suppose we have the following table for a
+    collection of 410 people::
+
+                  exposed   unexposed
+        cases         7         15
+        noncases     58        472
+
+    The question we ask is "Is exposure to the chemical associated with
+    increased risk of the disease?"
+
+    Compute the test, and first check the p-value.
+
+    >>> from scipy.stats.contingency import odds_ratio
+    >>> test = odds_ratio([[7, 15], [58, 472]])
+    >>> test.pvalue
+    0.009208708293019454
+
+    The p-value suggests that the apparent association of exposure
+    and cases is significant.
+
+    >>> test.odds_ratio
+    3.7836687705553564
+
+    The conditional odds ratio is larger than 1, which also suggests
+    there is an association.  We can compute 95% confidence interval
+    for the odds ratio:
+
+    >>> test.odds_ratio_ci(confidence_level=0.95)
+    ConfidenceInterval(low=1.251482913226682, high=10.363493716701287)
+
+    The 95% confidence interval for the conditional odds ratio is
+    approximately (1.25, 10.4), which is strong evidence that the
+    odds ratio is greater than 1.
+    """
+    if kind not in ['conditional', 'sample']:
+        raise ValueError("kind must be 'conditional' or 'sample'.")
+    if alternative not in ['two-sided', 'less', 'greater']:
+        raise ValueError("alternative must be 'two-sided', 'less' or "
+                         "'greater'.")
+
+    c = np.asarray(table)
+
+    if c.shape != (2, 2):
+        raise ValueError(f"Invalid shape {c.shape}. The input `table` must be "
+                         "of shape (2, 2).")
+
+    if not np.issubdtype(c.dtype, np.integer):
```

Comment:
That was for a scalar argument to `binomtest`.  Here we require an integer array.

Reply:
This is more of a question about that function - why did you write it if `np.issubdtype` does a good check? 
If I remember correctly, it will also check that the value is scalar and bounded on one side?